air-quality-ui/e2e/vAirify_summary_page.ts

```diff
@@ -0,0 +1,162 @@
+    this.page = page
+    this.apiForecast = apiForecast
+    this.apiSummary = apiSummary
+    this.title = this.page.locator('title')
+    this.scroller = this.page.locator('.ag-body-horizontal-scroll-viewport')
+    this.agCell = this.page.locator('role=gridcell')
+    this.allCells = this.page.locator('[role=gridcell]')
+    this.kampalaBtn = this.page.getByRole('link', { name: 'Kampala' })
+    this.abuDhabiBtn = this.page.getByRole('link', { name: 'Abu Dhabi' })
+    this.zurichBtn = this.page.getByRole('link', { name: 'Zurich' })
+    this.citiesBtn = this.page.getByRole('link', { name: 'Cities' })
+  }
+  async clickButton(buttonName: string) {
+    switch (buttonName) {
+      case 'Cities': {
+        await this.citiesBtn.click()
+        break
+      }
+      case 'Kampala': {
+        await this.kampalaBtn.click()
+        break
+      }
+      case 'Abu Dhabi': {
+        await this.abuDhabiBtn.click()
+        break
+      }
+      case 'Zurich': {
+        await this.zurichBtn.click()
+        break
+      }
+    }
+  }
+
+  async gotoSummaryPage() {
+    await this.page.goto('/city/summary')
+  }
+
+  async getTitle() {
+    const title = await this.page.title()
+    return title
+  }
+
+  async setupApiRoutes() {
+    await this.page.route('*/**/air-pollutant/forecast*', async (route) => {
+      await route.fulfill({ json: this.apiForecast })
+    })
+    await this.page.route(
+      '*/**/air-pollutant/measurements/summary*',
+      async (route) => {
+        await route.fulfill({ json: this.apiSummary })
+      },
+    )
+  }
+
+  async getColumnHeaderAndText(name: string, expectedText: string) {
+    const header = this.page.getByRole('columnheader', { name })
+    await header.waitFor({ state: 'visible' })
+    await expect(header).toHaveText(expectedText)
+  }
+
+  async scrollToRightmostPosition() {
+    await this.scroller.evaluate((element: HTMLElement) => {
+      element.scrollLeft = element.scrollWidth
+    })
+  }
+
+  async waitForGridVisible() {
+    await this.page.waitForSelector('.ag-root', { state: 'visible' })
+    await this.page.waitForSelector('.ag-header-cell', { state: 'visible' })
+  }
+
+  async checkNumbersHaveOneDecimalOnly(text: string) {
+    const number = parseFloat(text)
+    if (!isNaN(number)) {
+      const decimalPart = text.split('.')[1]
+      if (decimalPart && decimalPart.length > 1) {
+        throw new Error(`Number ${text} has more than one decimal place`)
+      }
+    }
+  }
+
+  async checkCellNumberFormat() {
+    const cells = await this.agCell.all()
+    for (const cell of cells) {
+      const cellText = await cell.textContent()
+      await this.checkNumbersHaveOneDecimalOnly(cellText || '')
+    }
+  }
+
+  async textSearch() {
+    const cells = await this.page.locator('.ag-cell').all()
+    let count = 0
+    for (const cell of cells) {
+      const cellText = await cell.innerText()
+      if (cellText.includes('Kyiv')) {
```

Comment:
Let's make 'kyiv' be passed in as a param and rename this to textSearchInCells or something

Reply:
It is done 💯